src/eins-hwinfo.c

```diff
@@ -394,18 +459,33 @@ record_computer_hwinfo (gpointer unused)
     emtr_event_recorder_record_event (emtr_event_recorder_get_default (),
                                       COMPUTER_HWINFO_EVENT,
                                       g_steal_pointer (&payload));
+  set_next_record_time ();
+
+  /* The interval of first record after each boot usually is not 24 hour. */
+  if (record_first_time ())
+    {
+      eins_boottimeout_add_seconds (RECORD_COMPUTER_HWINFO_INTERVAL_SECONDS,
+                                    record_computer_hwinfo,
+                                    NULL);
+      return G_SOURCE_REMOVE;
+    }
+
   return G_SOURCE_CONTINUE;
 }
 
 static void
 start_recording_record_computer_hwinfo (void)
 {
-  record_computer_hwinfo (NULL);
+  guint wait = get_wait_time_for_next_record ();
 
-  /* Record the computer hardware info every 24 hours. */
-  g_timeout_add_seconds (RECORD_COMPUTER_HWINFO_INTERVAL_SECONDS,
-                         record_computer_hwinfo,
-                         NULL);
+  /* If the wait time is less than 10mins, just skip the wait time. */
+  if (wait > 60u * 10u)
```

Comment:
I don't think this special case is correct. The intention is that this event is recorded at most once every 24 hours. This allows us to slice the data into 24-hour chunks and be reasonably sure we are not double-counting any systems. If we record the event every 23h50m, that property of the data isn't true any more.

Reply:
The idea is the wait time before first record after each boot, not the normal period.  I think about the PAYG allowable time before entering new credit.  However, I understand people will reboot again and again, even the credit is timeout.  So, next boot will catch it.  Therefore, we can remove this special case.